src/msw/dc.cpp

```diff
@@ -1439,6 +1439,9 @@ void wxMSWDCImpl::DoDrawRotatedText(const wxString& text,
     //     because it's not TrueType and so can't have non zero
     //     orientation/escapement
     wxFont font = m_font.IsOk() ? m_font : *wxSWISS_FONT;
+    if ( !m_font.IsOk() && m_window )
+        font.WXAdjustToPPI(m_window->GetDPI());
+
```

Comment:
This would result in a slightly larger diff, but I think it would be clearer to write this as
```cpp
wxFont font;
if ( m_font.IsOk() )
{
    font = m_font;
}
else // Use default font appropriate for rotated text.
{
    font = *wxSWISS_FONT;
    if ( m_window )
        font.WXAdjustToPPI(m_window->GetDPI());
}
```

BTW, I wonder if the comment about `DEFAULT_GUI_FONT` not being true type is still actually relevant (it was written in 4770df95a0a back in 2001 and certainly was true back then, but I must have tested it under something like Windows 2000...).